src/der.rs

```diff
@@ -105,22 +105,20 @@ pub(crate) fn expect_tag_and_get_value_limited<'a>(
 pub(crate) fn nested_limited<'a, R>(
     input: &mut untrusted::Reader<'a>,
     tag: Tag,
-    error: Error,
     decoder: impl FnOnce(&mut untrusted::Reader<'a>) -> Result<R, Error>,
     size_limit: usize,
 ) -> Result<R, Error> {
-    expect_tag_and_get_value_limited(input, tag, size_limit)?.read_all(error, decoder)
```

Comment:
for #38 i was thinking maybe we'd be passing in different errors based on call site, like `IssuingDistributionPoint::from_der` would pass in `ExcessData(IssuingDistributionPoint)` or something.

instead of more use of `BadDer`, could we have a new specific error which means "we didn't decode the entirety of the provided data"?

Reply:
Ah yes -- I think I've cleaned up all uses of `read_all()` to pass a more specific error.

Reply:
what do you think about using a `non_exhaustive` enum for the field of `Error::TrailingData`? (mainly for code size reasons)

Reply:
though I guess the code size justification only rings true if the whole end application avoids any dependency on `fmt::Debug for webpki::Error`, which seems a bit hopeful

Reply:
I guess the strings are a bit of a departure from the enum-only stuff. I suppose C FFI would be an issue, too, but I think rustls-ffi doesn't get exposed to `webpki::Error` directly? Okay, I'll make it an enum.

Reply:
>  but I think rustls-ffi doesn't get exposed to webpki::Error directly?

Yup, just the `rustls` `Error` enum :+1: